analyzers/tests/SonarAnalyzer.Test/Rules/MethodsShouldNotHaveIdenticalImplementationsTest.cs

```diff
@@ -22,36 +22,151 @@
+    private readonly VerifierBuilder builderVB = new VerifierBuilder<VB.MethodsShouldNotHaveIdenticalImplementations>();
+
+    [TestMethod]
+    public void MethodsShouldNotHaveIdenticalImplementations() =>
+        builderCS.AddPaths("MethodsShouldNotHaveIdenticalImplementations.cs").WithOptions(ParseOptionsHelper.FromCSharp8).Verify();
+
+    [DataTestMethod]
+    [DataRow("", "")]
+    [DataRow("where T: struct", "where T: struct")]
+    [DataRow("where T: class", "where T: class")]
+    [DataRow("where T: unmanaged", "where T: unmanaged")]
+    [DataRow("where T: new()", "where T: new()")]
+    [DataRow("where T: IEquatable<T>, IComparable",  "where T: System.IComparable, IEquatable<T>")]
+    public void MethodsShouldNotHaveIdenticalImplementations_MethodTypeParameters_NonCompliant(string constraint1, string constraint2) =>
+        builderCS.AddSnippet($$"""
+            using System;
+            public static class TypeConstraints
+            {
+                public static bool Compare1<T>(T? value1, T value2) {{constraint1}} // Secondary
+                {
+                    Console.WriteLine(value1);
+                    Console.WriteLine(value2);
+                    return true;
+                }
+
+                public static bool Compare2<T>(T? value1, T value2) {{constraint2}} // Noncompliant
+                {
+                    Console.WriteLine(value1);
+                    Console.WriteLine(value2);
+                    return true;
+                }
+            }
+            """).WithOptions(ParseOptionsHelper.FromCSharp9).Verify();
+
+    [CombinatorialData]
+    public void MethodsShouldNotHaveIdenticalImplementations_MethodTypeParameters_Compliant(
+        [DataValues("where T: struct", "where T: class")] string constraint1,
+        [DataValues("", "where T: unmanaged", "where T: new()")] string constraint2) =>
+        builderCS.AddSnippet($$"""
+            using System;
+            public static class TypeConstraints
+            {
+                public static bool Compare1<T>(T? value1, T value2) {{constraint1}}
+                {
+                    Console.WriteLine(value1);
+                    Console.WriteLine(value2);
+                    return true;
+                }
+
+                public static bool Compare2<T>(T? value1, T value2) {{constraint2}}
+                {
+                    Console.WriteLine(value1);
+                    Console.WriteLine(value2);
+                    return true;
+                }
+            }
+            """).WithOptions(ParseOptionsHelper.FromCSharp9).VerifyNoIssues();
+
+    [DataTestMethod]
+    [DataRow("", "")]
+    [DataRow("where TKey: struct", "where TKey: struct")]
+    [DataRow("where TKey: struct where TValue: class", "where TKey: struct where TValue: class")]
+    [DataRow("where TValue: class where TKey: struct", "where TKey: struct where TValue: class")]
```

Comment:
There are no compliant tests for methods with two type parameters. It would be good to test things like `where TKey: class where TValue: struct` vs `where TKey: struct where TValue: class` etc.